sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/changestreams/dao/BigtableChangeStreamAccessor.java

```diff
@@ -100,9 +101,30 @@ private static BigtableChangeStreamAccessor createAccessor(@NonNull BigtableConf
     dataSettingsBuilder.setInstanceId(instanceId);
     tableAdminSettingsBuilder.setInstanceId(instanceId);
 
-    if (appProfileId != null) {
-      dataSettingsBuilder.setAppProfileId(appProfileId);
-    }
+    String appProfileId = checkArgumentNotNull(bigtableConfig.getAppProfileId()).get();
```

Comment:
is `appProfileId` now mandatory?

Reply:
It is currently mandatory, we already had the `checkArgumentNotNull(bigtableConfig.getAppProfileId())` above. We are actually going to update this to default to the default app profile in a follow on though.